crates/cubecl-opt/src/block.rs

```diff
@@ -20,9 +20,9 @@ pub struct BasicBlock {
     /// The phi nodes that are required to be generated at the start of this block.
     pub phi_nodes: Rc<RefCell<Vec<PhiInstruction>>>,
     /// The variables written to by this block. Only set during the SSA transformation.
-    pub(crate) writes: HashSet<(u16, u8)>,
+    pub(crate) writes: HashSet<Id>,
     /// The live variables at the start of this block. Used for pruning phi nodes.
-    pub(crate) live_vars: HashSet<(u16, u8)>,
+    pub(crate) live_vars: HashSet<Id>,
     /// The dominance frontiers of this block (where phi nodes must be inserted).
```

Comment:
Maybe not for this PR, but could we replace the HashSet by a single Id / u32? Since we are allocating all Id in increasing order, we only need to know that how many there are.

Reply:
no, this is a list of variables that are currently live (their value is still needed), not all variables. So it's not contiguous. Same for writes, it's only variables written to in that block.